### ISSUE TITLE
`dotnet new -i` fails when I try to point to filesystem

### ISSUE THREAD

**Issue by wes566** [dotnet/templating#1361]
1. Clone https://github.com/dotnet/dotnet-template-samples
2. `cd dotnet-template-samples/01-basic-template/MyProject.Con`
3. `dotnet new -i .`

Fails to install the template, just gives me about a hundred lines of warnings like:
```
Restoring packages for /Users/user1/.templateengine/dotnetcli/v2.0.3/scratch/restore.csproj...
/usr/local/share/dotnet/sdk/2.0.3/NuGet.targets(102,5): warning : The folder '/usr/local/share/dotnet/sdk/NuGetFallbackFolder/.tools' contains an invalid version. [/Users/user1/.templateengine/dotnetcli/v2.0.3/scratch/restore.csproj]
```

and then at the end it gives me the error:
```
/Users/user1/.templateengine/dotnetcli/v2.0.3/scratch/restore.csproj : error NU1101: Unable to find package .. No packages exist with this id in source(s): /usr/local/share/dotnet/sdk/NuGetFallbackFolder, nuget.org
  Generating MSBuild file /Users/user1/.templateengine/dotnetcli/v2.0.3/scratch/obj/restore.csproj.nuget.g.props.
  Generating MSBuild file /Users/user1/.templateengine/dotnetcli/v2.0.3/scratch/obj/restore.csproj.nuget.g.targets.
  Restore failed in 596.09 ms for /Users/user1/.templateengine/dotnetcli/v2.0.3/scratch/restore.csproj.
```

This repros on both Windows and macOS for me.

```
compy386:MyProject.Con user1$ dotnet --info
.NET Command Line Tools (2.0.3)

Product Information:
 Version:            2.0.3
 Commit SHA-1 hash:  eb1d5ee318

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.0.3/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.3
  Build    : a9190d4a75f4a982ae4b4fa8d1a24526566c69df
```

**Comment by wes566** [comment]
This might be a dupe of #1306 as I have some private nuget feeds setup

**Comment by wes566** [comment]
Ok, I double checked on a different computer without a private nuget feed and I still hit this so I don't think this is a dupe of #1306

**Comment by mlorbetske** [comment]
@wes566 could you try doing `dotnet new -i .\` please? It seems like we're trying to treat `.` as a package name.

@seancpeters can we take a look at beefing up the logic that determines whether the thing to install is local a bit to take into account `.` and `..`? Looks like https://github.com/dotnet/templating/blob/rel/2.1.0-preview1/src/Microsoft.TemplateEngine.Cli/Installer.cs#L72 is indicating that `.` does not refer to something local.

**Comment by seancpeters** [comment]
This appears to be fixed by #1298
